packages/core/src/config/knapsack-pro-env.config.ts

```diff
@@ -74,14 +74,38 @@ export class KnapsackProEnvConfig {
     }
   }
 
-  public static get ciNodeTotal(): number | never {
-    if (process.env.KNAPSACK_PRO_CI_NODE_TOTAL) {
-      return parseInt(process.env.KNAPSACK_PRO_CI_NODE_TOTAL, 10);
+  private static logOverwrittenEnv(
+    envName: string,
+    envValue: string,
+    ciEnvValue: string,
+  ): void {
+    this.$knapsackProLogger.info(
+      `You have set the environment variable ${envName} to ${envValue} which could be automatically determined from the CI environment as ${ciEnvValue}.`,
+    );
+  }
+
+  private static envFor(
+    knapsackEnvName: string,
+    ciEnvFunction: CIProviderMethod,
+  ): string | boolean | undefined {
```

Comment:
Currently, TS understands that calling `envFor` returns either `string | boolean | undefined`. But `boolean` is only returned when calling `CIEnvConfig.fixedQueueSplit` in all the other cases it's only `string | undefined`.

Notice that having `boolean` in the return types obliges you to add `.toString()` in the `envFor` callers.

But you can teach TS to use the proper return types with:
```suggestion
  private static envFor<T extends CIProviderMethod>(
    knapsackEnvName: string,
    ciEnvFunction: T,
  ): (typeof CIEnvConfig)[T] {
```
and remove all the `.toString()` in the callers.